[CI] VectorSimilarityFunctionsIT class failing

**Build Scans:**
- [elasticsearch-periodic #11403 / openjdk24_checkpart3_java-matrix](https://gradle-enterprise.elastic.co/s/velhx2bpoj5uu)
- [elasticsearch-pull-request #102231 / part-3](https://gradle-enterprise.elastic.co/s/gsryn3xoealmy)
- [elasticsearch-pull-request #102230 / part-3](https://gradle-enterprise.elastic.co/s/wjmdg6bua6k74)
- [elasticsearch-pull-request #102228 / part-3](https://gradle-enterprise.elastic.co/s/bhgysecvlwqme)
- [elasticsearch-intake #30801 / part3](https://gradle-enterprise.elastic.co/s/tszmqhgmhmhnc)
- [elasticsearch-pull-request #102145 / part-3](https://gradle-enterprise.elastic.co/s/anz327h6mj2cq)
- [elasticsearch-pull-request #102119 / part-3](https://gradle-enterprise.elastic.co/s/svz56wxcm77rs)
- [elasticsearch-pull-request #102101 / part-3](https://gradle-enterprise.elastic.co/s/fx7u4622abl5i)
- [elasticsearch-pull-request #102061 / part-3](https://gradle-enterprise.elastic.co/s/gmhjndhscoxwc)
- [elasticsearch-pull-request #102050 / part-3](https://gradle-enterprise.elastic.co/s/rxorpn7ayds6e)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:internalClusterTest" --tests "org.elasticsearch.xpack.esql.vector.VectorSimilarityFunctionsIT" -Dtests.method="testSimilarityWithOneDimVector {functionName=v_cosine similarityFunction=org.elasticsearch.xpack.esql.expression.function.vector.CosineSimilarity$1@424ebba3 elementType=float}" -Dtests.seed=69F510695D801E11 -Dtests.locale=cv -Dtests.timezone=Africa/Nairobi -Druntime.java=24
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.vector.VectorSimilarityFunctionsIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('org.elasticsearch.xpack.esql.vector.VectorSimilarityFunctionsIT'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: timeout
```

**Issue Reasons:**
- [main] 26 failures in class org.elasticsearch.xpack.esql.vector.VectorSimilarityFunctionsIT (4.9% fail rate in 530 executions)
- [main] 17 failures in step part-3 (5.5% fail rate in 311 executions)
- [main] 3 failures in step part3 (3.5% fail rate in 85 executions)
- [main] 2 failures in step amazonlinux-2023_platform-support-aws (40.0% fail rate in 5 executions)
- [main] 17 failures in pipeline elasticsearch-pull-request (5.5% fail rate in 307 executions)
- [main] 2 failures in pipeline elasticsearch-periodic (33.3% fail rate in 6 executions)
- [main] 2 failures in pipeline elasticsearch-periodic-platform-support (33.3% fail rate in 6 executions)
- [main] 3 failures in pipeline elasticsearch-intake (3.5% fail rate in 85 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -477,10 +477,6 @@ tests:
 - class: org.elasticsearch.xpack.ilm.CCRIndexLifecycleIT
   method: testCcrAndIlmWithRollover {targetCluster=FOLLOWER}
   issue: https://github.com/elastic/elasticsearch/issues/137528
-- class: org.elasticsearch.xpack.esql.vector.VectorSimilarityFunctionsIT
-  method: testSimilarityWithOneDimVector {functionName=v_cosine
-    similarityFunction=org.elasticsearch.xpack.esql.expression.function.vector.CosineSimilarity$1@6f152006 elementType=byte}
-  issue: https://github.com/elastic/elasticsearch/issues/137552
 
 # Examples:
 #

--- a/x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/vector/VectorSimilarityFunctionsIT.java
+++ b/x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/vector/VectorSimilarityFunctionsIT.java
@@ -158,7 +158,7 @@ public class VectorSimilarityFunctionsIT extends AbstractEsqlIntegTestCase {
         var query = String.format(Locale.ROOT, """
                 FROM test
                 | EVAL similarity = %s(one_dim_vector, %s)
-                | KEEP left_vector, similarity
+                | KEEP one_dim_vector, similarity
             """, functionName, randomVector);
         try (var resp = run(query)) {
             List<List<Object>> valuesList = EsqlTestUtils.getValuesList(resp);

--- a/x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/vector/VectorSimilarityFunctionsIT.java
+++ b/x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/vector/VectorSimilarityFunctionsIT.java
@@ -163,7 +163,7 @@ public class VectorSimilarityFunctionsIT extends AbstractEsqlIntegTestCase {
         try (var resp = run(query)) {
             List<List<Object>> valuesList = EsqlTestUtils.getValuesList(resp);
             valuesList.forEach(values -> {
-                List<Number> vecAsList = ((List<Number>) values.get(0));
+                List<Number> vecAsList = values.get(0) == null ? null : List.of((Number) values.get(0));
                 Double similarity = (Double) values.get(1);
                 if (vecAsList == null || randomVector == null) {
                     assertNull(similarity);

--- a/x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/vector/VectorSimilarityFunctionsIT.java
+++ b/x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/vector/VectorSimilarityFunctionsIT.java
@@ -19,7 +19,6 @@ import org.elasticsearch.core.TimeValue;
 import org.elasticsearch.index.mapper.vectors.DenseVectorFieldMapper;
 import org.elasticsearch.index.mapper.vectors.DenseVectorFieldMapper.ElementType;
 import org.elasticsearch.plugins.Plugin;
-import org.elasticsearch.test.junit.annotations.TestLogging;
 import org.elasticsearch.xcontent.XContentBuilder;
 import org.elasticsearch.xcontent.XContentFactory;
 import org.elasticsearch.xpack.esql.EsqlTestUtils;
@@ -42,7 +41,7 @@ import java.util.Set;
 import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
 import static org.hamcrest.CoreMatchers.containsString;
 
-@TestLogging(value = "org.elasticsearch.xpack.esql:TRACE", reason = "debug")
+//@TestLogging(value = "org.elasticsearch.xpack.esql:TRACE", reason = "debug")
 public class VectorSimilarityFunctionsIT extends AbstractEsqlIntegTestCase {
 
     private List<List<Number>> leftVectors;
@@ -155,21 +154,24 @@ public class VectorSimilarityFunctionsIT extends AbstractEsqlIntegTestCase {
 
     @SuppressWarnings("unchecked")
     public void testSimilarityWithOneDimVector() {
-        final float oneDimFloat = randomFloat();
         var randomVector = randomVector(1);
         var query = String.format(Locale.ROOT, """
-                ROW left_vector = to_dense_vector(%s)
-                | EVAL similarity = %s(left_vector, %f)
+                FROM test
+                | EVAL similarity = %s(one_dim_vector, %s)
                 | KEEP left_vector, similarity
-            """, randomVector, functionName, oneDimFloat);
+            """, functionName, randomVector);
         try (var resp = run(query)) {
             List<List<Object>> valuesList = EsqlTestUtils.getValuesList(resp);
             valuesList.forEach(values -> {
-                float[] left = new float[] { (float) values.get(0) };
+                List<Number> vecAsList = ((List<Number>) values.get(0));
                 Double similarity = (Double) values.get(1);
-                assertNotNull(similarity);
-                float expectedSimilarity = similarityFunction.calculateSimilarity(left, new float[] { oneDimFloat });
-                assertEquals(expectedSimilarity, similarity, 0.0001);
+                if (vecAsList == null || randomVector == null) {
+                    assertNull(similarity);
+                } else {
+                    assertNotNull(similarity);
+                    double expectedSimilarity = calculateSimilarity(similarityFunction, randomVector, vecAsList);
+                    assertEquals(expectedSimilarity, similarity, 0.0001);
+                }
             });
         }
     }
@@ -251,7 +253,11 @@ public class VectorSimilarityFunctionsIT extends AbstractEsqlIntegTestCase {
     }
 
     public void testDifferentDimensions() {
-        var randomVector = randomVector(randomValueOtherThan(numDims, () -> randomIntBetween(32, 64) * 2));
+        // edge case where this might not throw is if all `left_vector` are null, but the chance is (hopefully!) low enough to ignore
+        var randomVector = randomValueOtherThan(
+            null,
+            () -> randomVector(randomValueOtherThan(numDims, () -> randomIntBetween(32, 64) * 2))
+        );
         var query = String.format(Locale.ROOT, """
                 FROM test
                 | EVAL similarity = %s(left_vector, %s)
@@ -309,8 +315,9 @@ public class VectorSimilarityFunctionsIT extends AbstractEsqlIntegTestCase {
         for (int i = 0; i < numDocs; i++) {
             List<Number> leftVector = randomVector();
             List<Number> rightVector = randomVector();
+            List<Number> oneDimVector = randomVector(1);
             docs[i] = prepareIndex("test").setId("" + i)
-                .setSource("id", String.valueOf(i), "left_vector", leftVector, "right_vector", rightVector);
+                .setSource("id", String.valueOf(i), "left_vector", leftVector, "right_vector", rightVector, "one_dim_vector", oneDimVector);
             leftVectors.add(leftVector);
         }
 
@@ -347,6 +354,7 @@ public class VectorSimilarityFunctionsIT extends AbstractEsqlIntegTestCase {
             .endObject();
         createDenseVectorField(mapping, "left_vector");
         createDenseVectorField(mapping, "right_vector");
+        createDenseVectorField(mapping, "one_dim_vector");
         mapping.endObject().endObject();
         Settings.Builder settingsBuilder = Settings.builder()
             .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)

